File: Source/Backend/mekongocop-server/src/main/java/com/mekongocop/mekongocopserver/dto/UserProfileDTO.java
package com.mekongocop.mekongocopserver.dto;

import jakarta.validation.constraints.Size;
import org.jetbrains.annotations.NotNull;

import java.time.LocalDate;

public class UserProfileDTO {
    public int profileId;

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public int user_id;
    public String full_name;

    @NotNull
    public LocalDate birthday;
    @NotNull
    public char sex;
    @Size(max=200, message = "Max length 200")
    public String bio;

}

File: Source/Frontend/Flutter-Ecommerce/lib/dto/user_profile_dto.dart
class UserProfileDTO {
  final String fullName;
  final String profileId;
  final String sex;
  final String bio;
  final String birthday;
  final String userId;

  UserProfileDTO({
    required this.fullName,
    required this.profileId,
    required this.sex,
    required this.bio,
    required this.birthday,
    required this.userId,
  });

  // Hàm ánh xạ từ JSON
  factory UserProfileDTO.fromJson(Map<String, dynamic> json) {
    return UserProfileDTO(
      fullName: json['full_name'] ?? '', // Ánh xạ từ 'full_name' thành 'fullName'
      profileId: json['profileId'] ?? '',
      sex: json['sex'] ?? '',
      bio: json['bio'] ?? '',
      birthday: json['birthday'] ?? '',
      userId: json['user_id'] ?? '',
    );
  }

  // Hàm chuyển đối tượng thành JSON
  Map<String, dynamic> toJson() {
    return {
      'full_name': fullName, // Ánh xạ từ 'fullName' thành 'full_name'
      'profileId': profileId,
      'sex': sex,
      'bio': bio,
      'birthday': birthday,
      'user_id': userId,
    };
  }
}
